src/modules/job-exec/job-exec.c

```diff
@@ -1440,6 +1440,33 @@ static int unload_implementations (struct job_exec_ctx *ctx)
     return 0;
 }
 
+static void config_reload_cb (flux_t *h,
+                              flux_msg_handler_t *mh,
+                              const flux_msg_t *msg,
+                              void *arg)
+{
+    const flux_conf_t *conf;
+    struct exec_implementation *impl;
+    int i = 0;
+
+    if (flux_conf_reload_decode (msg, &conf) < 0)
+        goto error;
+
+    while ((impl = implementations[i]) && impl->name) {
+        if (impl->reload) {
+            if ((*impl->reload) (h, conf) < 0)
+                goto error;
+        }
+        i++;
+    }
+    if (flux_respond (h, msg, NULL) < 0)
+        flux_log_error (h, "error responding to config-reload request");
+    return;
+error:
+    if (flux_respond_error (h, msg, errno, NULL) < 0)
+        flux_log_error (h, "error responding to config-reload request");
```

Comment:
Returning a human readable error here would be helpful as it is displayed by `flux config reload`.

Add `flux_error_t *` to `impl->reload` and `flux_conf_reload_decode()` perhaps?